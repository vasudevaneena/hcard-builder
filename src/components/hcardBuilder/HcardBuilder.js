import React from "react";

import HcardButton from "../hcardButton/HcardButton";
import HcardPreview from "../hcardPreview/HcardPreview";
import hcardBuilderHook from "../../hooks/HcardBuilderHook";
import "./HcardBuilder.scss";
import * as constants from "../../constants";

const HcardBuilder = () => {
  const { inputs, handleInputChange } = hcardBuilderHook();
  return (
    <form>
      <div className="container">
        <div className="row">
           <div className="col-md-1 col-xs-1"></div> 
          <div className="col-md-5 col-xs-5 hcard-builder">
            <div className="inner-padding">
              <h4 className="heading">hCard Builder</h4>

              <div className="underline col-md-10 top">
                <label className="label-color">
                  {constants.PersonalDetails}
                </label>
              </div>
              <div className="row">
                <div className="col-md-5">
                  <div>
                    <label htmlFor="givenname">{constants.GivenName}</label>
                  </div>
                  <div>
                    <input
                      id="givenname"
                      type="text"
                      name="givenname"
                      onChange={handleInputChange}
                      value={inputs.givenname || ""}
                    ></input>
                  </div>
                </div>
                <div className="col-md-5">
                  <div>
                    <label htmlFor="surname">{constants.Surname}</label>
                  </div>
                  <div>
                    <input
                      id="surname"
                      type="text"
                      name="surname"
                      onChange={handleInputChange}
                      value={inputs.surname || ""}
                    ></input>
                  </div>
                </div>
              </div>
              <div className="row">
                <div className="col-md-5">
                  <div>
                    <label htmlFor="email">{constants.Email}</label>
                  </div>
                  <div>
                    <input
                      id="email"
                      type="text"
                      name="email"
                      onChange={handleInputChange}
                      value={inputs.email || ""}
                    ></input>
                  </div>
                </div>
                <div className="col-md-5">
                  <div>
                    <label htmlFor="phone">{constants.Phone}</label>
                  </div>
                  <div>
                    <input
                      id="phone"
                      type="text"
                      name="phone"
                      onChange={handleInputChange}
                      value={inputs.phone || ""}
                    ></input>
                  </div>
                </div>
              </div>
              <div className="underline col-md-10 top">
                <label className="label-color">{constants.Address}</label>
              </div>
              <div className="row">
                <div className="col-md-5">
                  <div>
                    <label htmlFor="housename">{constants.HouseName}</label>
                  </div>
                  <div>
                    <input
                      id="housename"
                      type="text"
                      name="housename"
                      onChange={handleInputChange}
                      value={inputs.housename || ""}
                    ></input>
                  </div>
                </div>
                <div className="col-md-5">
                  <div>
                    <label htmlFor="street">{constants.Street}</label>
                  </div>
                  <div>
                    <input
                      id="street"
                      type="text"
                      name="street"
                      onChange={handleInputChange}
                      value={inputs.street || ""}
                      
                    ></input>
                  </div>
                </div>
              </div>
              <div className="row">
                <div className="col-md-5">
                  <div>
                    <label htmlFor="suburb">{constants.Suburb}</label>
                  </div>
                  <div>
                    <input
                      id="suburb"
                      type="text"
                      name="suburb"
                      onChange={handleInputChange}
                      value={inputs.suburb || ""}
                      
                    ></input>
                  </div>
                </div>
                <div className="col-md-5">
                  <div>
                    <label htmlFor="state">{constants.State}</label>
                  </div>
                  <div>
                    <input
                      id="state"
                      type="text"
                      name="state"
                      onChange={handleInputChange}
                      value={inputs.state || ""}
                      
                    ></input>
                  </div>
                </div>
              </div>

              <div className="row">
                <div className="col-md-5">
                  <div>
                    <label htmlFor="postcode">{constants.Postcode}</label>
                  </div>
                  <div>
                    <input
                      id="postcode"
                      type="text"
                      name="postcode"
                      onChange={handleInputChange}
                      value={inputs.postcode || ""}
                      
                    ></input>
                  </div>
                </div>
                <div className="col-md-5">
                  <div>
                    <label htmlFor="country">{constants.Country}</label>
                  </div>
                  <div>
                    <input
                      id="country"
                      type="text"
                      name="country"
                      onChange={handleInputChange}
                      value={inputs.country || ""}
                      
                    ></input>
                  </div>
                </div>
              </div>
              <div className="row top">
                <HcardButton value="Upload Avatar" />
                <HcardButton value="Create hCard" />
              </div>
            </div>
          </div>
          <div className="col-md-1 col-xs-1"></div>
          <div className="col-md-5 col-xs-5 preview-top">
       
            <HcardPreview props={inputs} />
          </div>
        </div>
      </div>
     </form> 
  );
};

export default HcardBuilder;
