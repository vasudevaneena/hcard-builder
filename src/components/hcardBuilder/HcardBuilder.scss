@import "../../styles/config.scss";
.hcard-builder {
  background-color: $color-white;
  padding: 10px 20px 5px 16px;
  margin-top: 5px;
  .inner-padding {
    padding: 20px ;
  }

  .top {
    margin-top: 25px;
  }
  
  .heading {
    color: $color-heading;
  }

  .label-color {
    color: $color-label;
  }
  input {
    width: 155px;
    -webkit-appearance: none;
    line-height: 1em;
    padding: 0.47857143em;
    font-size: 1em;
    background: $color-white;
    border: 1px solid rgba(34, 36, 38, 0.15);
    color: rgba(0, 0, 0, 0.87);
    border-radius: 0.28571429rem;
    box-shadow: 0 0 0 0 transparent inset;
    transition: color 0.1s ease, border-color 0.1s ease;
  }
  @media (min-width: 992px) and (max-width: 1199px) {
    input {
      width: 124px !important;
      font-size: 0.8em;
    }
  }
  @media (min-width: 768px) and (max-width: 992px) {
    input {
      width: 85px !important;
      font-size: 0.8em;
    }
  }
}
.preview-top {
  margin-top: 85px;
}
