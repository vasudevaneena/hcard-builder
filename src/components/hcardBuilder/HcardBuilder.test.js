import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";
import HcardBuilder from "./HcardBuilder";

configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
  wrapper = mount(<HcardBuilder />);
});


describe("<Form /> rendering", () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
});
    it("should render one <UserImage>", () => {
      expect(wrapper.find("UserImage")).toHaveLength(1);
    });
    it("should render child component  <HcardPreview>", () => {
      expect(wrapper.find("HcardPreview")).toHaveLength(1);
    });
    it("should render one <form>", () => {
      expect(wrapper.find("form")).toHaveLength(1);
    });
 
    it("should render 2 <button>s", () => {
      expect(wrapper.find("button")).toHaveLength(2);
    });
  });

  describe("<Form /> interactions", () => {
    it("should update the email input field", async () => {
        
      //console.log(wrapper.debug());   
      wrapper.update();
      await wrapper
        .find("#email")
        .simulate("change", { target: {name:"email", value: "xyz@gmail.com" } });
       
      expect(wrapper.find("#email").props().value).toEqual("xyz@gmail.com");
    });
  
    
  });