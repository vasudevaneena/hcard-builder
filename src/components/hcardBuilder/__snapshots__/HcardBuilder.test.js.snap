// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> rendering renders correctly 1`] = `ReactWrapper {}`;
